### ISSUE TITLE
TypeError: op.<locals>.wrap() takes 1 positional argument but 2 were given

### ISSUE THREAD

**Issue by SalamanderXing** [wandb/weave#1584]
Adding `weave.op` causes my code to break. I prepared a notebook to reproduce the issue:

https://colab.research.google.com/drive/1XRU11SIytHXqiO4mV3CYy5JePw1B2YJC?usp=sharing

```
TypeError                                 Traceback (most recent call last)
[<ipython-input-16-be0796def7f5>](https://localhost:8080/#) in <cell line: 1>()
      1 while True:
      2     new_msg = input("Your msg:")
----> 3     await graph.run(dict(chat=[HumanMessage(content=new_msg)]))
      4 

[/usr/local/lib/python3.10/dist-packages/state_graph/__init__.py](https://localhost:8080/#) in run(self, input)
    412                 return self.context, False
    413 
--> 414             context_update = await self.current_node.run(self.context)
    415             self.context = update_context(self.context, context_update)
    416 

TypeError: op.<locals>.wrap() takes 1 positional argument but 2 were given
```

**Comment by tssweeney** [comment]
Hello @SalamanderXing - thank you very much for your bug report and reproduction. To fix your code, please call the wrapper:

```
@weave.op # change this to `@weave.op()`
async def run(self, context: Context):
   ...
```

so the resulting code should be

```
@weave.op()
async def run(self, context: Context):
   ...
```

Note: in colab, you have to call `weave.init()` in any cell that uses `await` due to an odd colab bug. So add `weave.init` to the cell with the while loop as well.

---

Note to team: this should be broken into 2 tickets:
1. When wrapping a method with our wrapper, we should consider making the calling optional (especially since it takes no arguments). Or at the very least give a good error message.
2. We should attempt to patch the colab bug

**Comment by SalamanderXing** [comment]
Hi @tssweeney, thank you for your reply That indeed fixed that issue. 
Still, I'm getting another error trying to apply Weave in my source code, I modified the [notebook](https://colab.research.google.com/drive/1XRU11SIytHXqiO4mV3CYy5JePw1B2YJC?usp=sharing#scrollTo=uVdqypRSRmgO) to reproduce that one too. It's most likely due to pydantic v1 vs v2 mismatch.
The error is different but related to #1579. Please let me know if I should open a new issue.

**Comment by tssweeney** [comment]
Hi @SalamanderXing - thank you for the repro. There is something definitely odd going on here. Let me try to summarize:
1. As discussed in https://github.com/wandb/weave/issues/1579, you cannot use Pydantic V1 as type annotations on Weave Objects.
2. You have to "call" the `weave.op` decorator. My colleague @jamie-rasmussen fixed this in the next release here: https://github.com/wandb/weave/commit/2818f451c2161c4a92efe1e744e7f2cee2b18339
3. In your case, the last cell in your notebook throws the error your quote above. Note: if i as `weave.init` to that cell (per my comment above), it works every other time. This is very odd. 

I am going to create two internal tickets. Note: these links are not publicly accessible.

1. Support PydanticV1 as properties: https://wandb.atlassian.net/browse/WB-18538
2. Investigate consistent failing in this case (could be pydantic... also could be the async/await in notebooks): https://wandb.atlassian.net/browse/WB-18539

We will update this thread upon further information. Please allow us a few days to investigate.

**Comment by tssweeney** [comment]
Reopening after accidentally hitting "close" button.

**Comment by tssweeney** [comment]
Hi @SalamanderXing I just merged in some changes that should unblock you. In particular, fixed the PydanticV1 issue you described as well as another issue I noticed while working with your repro. 

Open items still (although both of these have workarounds):
* Using PydanticV1 types in annotations
* using `await` in a colab removes the `weave.init` status upon cell completion, requiring you to re-init after such calls

**Comment by SalamanderXing** [comment]
Thanks @tssweeney. In the meantime I migrated away from langchain so im now using only Pydantic v2 :)

**Comment by andrewtruong** [comment]
Closing as stale -- if this is still an issue, please re-open
